File: 003 - Largest prime factor/3.c
#include <stdio.h>

main(){
 long long int a = 600851475143;
 int i, b = 0;

for(i=3;;i+=2){
  while ((a % i)==0){
   a /= i;
   b = i;
  }

  if (a == 1)
   break;

}
 printf ("Highest prime factor of 600851475143 is : %d\n", b);
}

File: project_euler/003 - Largest prime factor/3.py
max = 0
num = 600851475143L
n=3L

while True:
    while num%n==0:
        num /= n
        max = n
    if num==1:
        break
    n+=2L

print "The largest prime factor of the number 600851475143: ", max
    
